src/os/bluestore/BlueStore.cc

```diff
@@ -5375,13 +5375,14 @@ int BlueStore::_write_bdev_label(CephContext *cct,
   z.zero();
   bl.append(std::move(z));
 
-  int fd = TEMP_FAILURE_RETRY(::open(path.c_str(), O_WRONLY|O_CLOEXEC));
+  int fd = TEMP_FAILURE_RETRY(::open(path.c_str(), O_WRONLY|O_CLOEXEC|O_DIRECT));
```

Comment:
From https://man7.org/linux/man-pages/man2/open.2.html this flag "Try to minimize cache effects of the I/O  ... but does not give the guarantees of the O_SYNC flag that data and necessary metadata are transferred"
Given that comment wouldn't be better to use O_SYNC flag and/or call fsync before closing the file descriptor?

Reply:
I am a bit confused. Do you advise that we should use O_SYNC here? In this func Line  5392, it does use fsync() to flush the data to device.

Reply:
ah.. my bad - there is fsync indeed. O_DIRECT + fsync should be sufficient then...